gctscript/modules/ta/indicators/macd.go

```diff
@@ -0,0 +1,73 @@
+package indicators
+
+import (
+	"fmt"
+
+	objects "github.com/d5/tengo/v2"
+	"github.com/thrasher-corp/go-talib/indicators"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/modules"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/wrappers/validator"
+)
+
+var MACDModule = map[string]objects.Object{
+	"calculate": &objects.UserFunction{Name: "calculate", Value: macd},
+}
+
+func macd(args ...objects.Object) (objects.Object, error) {
+	if len(args) != 4 {
+		return nil, objects.ErrWrongNumArguments
+	}
+
+	ohlcvInput := objects.ToInterface(args[0])
+	ohlcvInputData, valid := ohlcvInput.([]interface{})
+	if !valid {
+		return nil, fmt.Errorf(modules.ErrParameterConvertFailed, OHLCV)
+	}
+
+	var ohlcvClose []float64
+	for x := range ohlcvInputData {
+		switch t := ohlcvInputData[x].(type) {
+		case []interface{}:
+			value, err := toFloat64(t[4])
+			if err != nil {
+				return nil, err
+			}
+			ohlcvClose = append(ohlcvClose, value)
+		default:
+			return nil, fmt.Errorf(modules.ErrParameterConvertFailed, OHLCV)
+		}
+	}
+
+	inFastPeriod, ok := objects.ToInt(args[1])
+	if !ok {
+		return nil, fmt.Errorf(modules.ErrParameterConvertFailed, inFastPeriod)
+	}
+	inSlowPeriod, ok := objects.ToInt(args[2])
+	if !ok {
+		return nil, fmt.Errorf(modules.ErrParameterConvertFailed, inSlowPeriod)
+	}
+	inTimePeriod, ok := objects.ToInt(args[3])
+	if !ok {
+		return nil, fmt.Errorf(modules.ErrParameterConvertFailed, inTimePeriod)
+	}
+
+	var macd, macdSignal, macdHist []float64
+	if validator.IsTestExecution.Load() != true {
```

Comment:
we still changing this?

Reply:
After playing around with different setups i decided against changing it everything else just added more complexity

the wrapper/module setup was originally added to stop circular imports from engine package and it is very clunky in the first place

I might end up rewording "IsTestExecution" at some point in the future as that is really my issue with it since its not used for testing for validating as well

Reply:
👍 Thanks for the explanation.